specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json

```diff
@@ -0,0 +1,5291 @@
+          "description": "Blob restore status",
+          "readOnly": true
+        },
+        "allowBlobPublicAccess": {
+          "type": "boolean",
+          "x-ms-client-name": "AllowBlobPublicAccess",
+          "description": "Allow or disallow public access to all blobs or containers in the storage account. The default interpretation is true for this property."
+        },
+        "minimumTlsVersion": {
+          "type": "string",
+          "enum": [
+            "TLS1_0",
+            "TLS1_1",
+            "TLS1_2"
+          ],
+          "x-ms-enum": {
+            "name": "MinimumTlsVersion",
+            "modelAsString": true
+          },
+          "description": "Set the minimum TLS version to be permitted on requests to storage. The default interpretation is TLS 1.0 for this property."
+        },
+        "allowSharedKeyAccess": {
+          "type": "boolean",
+          "description": "Indicates whether the storage account permits requests to be authorized with the account access key via Shared Key. If false, then all requests, including shared access signatures, must be authorized with Azure Active Directory (Azure AD). The default value is null, which is equivalent to true."
+        },
+        "isNfsV3Enabled": {
+          "type": "boolean",
+          "x-ms-client-name": "EnableNfsV3",
+          "description": "NFS 3.0 protocol support enabled if set to true."
+        },
+        "allowCrossTenantReplication": {
+          "type": "boolean",
+          "description": "Allow or disallow cross AAD tenant object replication. The default interpretation is true for this property."
+        },
+        "defaultToOAuthAuthentication": {
+          "type": "boolean",
+          "description": "A boolean flag which indicates whether the default authentication is OAuth or not. The default interpretation is false for this property."
+        },
+        "publicNetworkAccess": {
+          "type": "string",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "name": "PublicNetworkAccess",
+            "modelAsString": true
+          },
+          "description": "Allow or disallow public network access to Storage Account. Value is optional but if passed in, must be 'Enabled' or 'Disabled'."
+        },
+        "immutableStorageWithVersioning": {
+          "$ref": "#/definitions/ImmutableStorageAccount",
+          "x-ms-client-name": "ImmutableStorageWithVersioning",
+          "description": "The property is immutable and can only be set to true at the account creation time. When set to true, it enables object level immutability for all the containers in the account by default."
+        },
+        "allowedCopyScope": {
+          "type": "string",
+          "enum": [
+            "PrivateLink",
+            "AAD"
+          ],
+          "x-ms-enum": {
+            "name": "AllowedCopyScope",
+            "modelAsString": true
```

Comment:
Why user `"modelAsString": true`?
If there are any possible new value to the enum in the future, use `"modelAsString": false` can avoid breaking.

Reply:
@raomanus, could you please check?

Reply:
@blueww, why "modelAsString": false can avoid SDK breaking? I am seeing other enums are modeled as strings.

Reply:
My fault. Yes, should use `"modelAsString": true` to avoid SDK breaking.